File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,bool> m;
        for (const int& n : nums) {
            if (m[n]) return true;
            else m[n] = true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
from itertools import accumulate
import operator

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = [1] + list(accumulate(nums[:-1],operator.mul))
        right = [1] + list(accumulate(nums[::-1][:-1],operator.mul))
        return [lacc * racc for lacc, racc in zip(left,reversed(right))]
        